### PR TITLE
Add options, trace and patch method

### PR DESCRIPTION
**Pull request by laurentL**

### CODE REVIEW

**Comment by laurentL** [comment]
errors in python 3
/bin/bash: line 20: /opt/python/cp33-cp33m/bin/pip: No such file or directory
/Users/travis/.travis/job_stages: line 339: shell_session_update: command not found

File: src/geventhttpclient/client.py [hunk 1]
```diff
@@ -211,24 +255,50 @@ def request(self, method, request_uri, body=b"", headers={}):
                 response._sent_request = request
                 return response
 
-    def get(self, request_uri, headers={}):
+    def get(self, request_uri, headers=None):
+        if headers is None:
```
Comment (ml31415) [inline src/geventhttpclient/client.py]: As the None handling is perfectly covered in `request`, I'd leave out all this redundant code here.

Comment (laurentL) [reply src/geventhttpclient/client.py]: Default argument value is mutable.
you could read this: https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html

Comment (ml31415) [reply src/geventhttpclient/client.py]: I'm all for changing the default argument to None. It's just the `if`-clause that can be omitted, as None gets converted to a dict anyways in the `request` method.

Comment (laurentL) [reply src/geventhttpclient/client.py]: ok, excuse me..

**Pull request merged**